flink-runtime/src/main/java/org/apache/flink/runtime/state/CheckpointableKeyedStateBackend.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.state;
+
+import java.io.Closeable;
+
+/**
+ * Interface that combines both, the {@link KeyedStateBackend} interface, which encapsulates methods
+ * responsible for keyed state management and the {@link SnapshotStrategy} which tells the system
+ * how to snapshot the underlying state.
+ *
+ * <p><b>NOTE:</b> Implementations of this class can be notified about a checkpoint result it can additionally implement
+ * the {@link CheckpointListener} interface.
+ *
+ * @param <K> The key by which state is keyed.
+ */
```

Comment:
```suggestion
 * <p><b>NOTE:</b> State backends that need to be notified of completed checkpoints can additionally implement
 * the {@link CheckpointListener} interface.
 *
 * @param <K> Type of the key by which state is keyed.
 */
```
Formatting is probably not correct after my changes but I think the previous sentence was not grammatically correct.